src/app/Http/Controllers/Admin/UniquePageCrudController.php

```diff
@@ -0,0 +1,245 @@
+
+        $this->checkForTemplatesAndUniquePagesNotDistinct();
+
+        /*
+        |--------------------------------------------------------------------------
+        | BASIC CRUD INFORMATION
+        |--------------------------------------------------------------------------
+        */
+        $this->crud->setModel($modelClass);
+        // Don't set route or entity names here. These depend on the page you are editing
+
+        // unique pages cannot be created nor deleted
+        $this->crud->denyAccess(['list', 'create', 'delete']);
+
+        if (config('backpack.pagemanager.unique_page_revisions')) {
+            $this->crud->allowAccess('revisions');
+        }
+    }
+
+    /**
+     * Edit the unique page retrieved by slug.
+     *
+     * @param string $slug
+     * @return Response
+     */
+    public function uniqueEdit($slug)
+    {
+        $model = $this->crud->model;
+        $entry = $model::findBySlug($slug);
+
+        if (! $entry) {
+            $entry = $this->createMissingPage($slug);
+        }
+
+        $this->uniqueSetup($entry);
+
+        return parent::edit($entry->id);
+    }
+
+    /**
+     * Update the unique page.
+     *
+     * @param string $slug
+     * @param int $id
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function update($slug, $id)
+    {
+        $model = $this->crud->model;
+        $entry = $model::findBySlugOrFail($slug);
+
+        $this->uniqueSetup($entry);
+
+        return parent::updateCrud();
+    }
+
+    /**
+     * Set the crud route.
+     *
+     * @param $slug
+     */
+    public function setRoute($slug)
+    {
+        $this->crud->setRoute(config('backpack.base.route_prefix').'/unique/'.$slug);
+    }
+
+    /**
+     * Populate the update form with basic fields that all pages need.
+     *
+     * @param Model $page
+     */
+    public function addDefaultPageFields($page)
+    {
+        $fields = [
+            [
+                'name'  => 'buttons',
+                'type'  => 'custom_html',
+                'value' => $this->buttons($page),
+            ],
+            [
+                'name' => 'template',
+                'type' => 'hidden',
+            ],
+            [
+                'name' => 'name',
+            ],
+            [
+                'name' => 'title',
+            ],
+            [
+                'name' => 'slug',
+                'type' => 'hidden',
+            ],
+        ];
+
+        $this->crud->addFields($fields);
+    }
+
+    /**
+     * Build the buttons html for the edit form.
+     *
+     * @param Model $page
+     * @return string
+     */
+    public function buttons($page)
+    {
+        $openButton = $page->getOpenButton();
+        $revisionsButton = view('crud::buttons.revisions', ['crud' => $this->crud, 'entry' => $page]);
+
+        return $openButton.' '.$revisionsButton;
+    }
+
+    /**
+     * Create the page by slug.
```

Comment:
This and the function name are slightly out of sync don't we think?

Reply:
this is why I don't linke comments... function name says it all.. will fix